### PR TITLE
Refactor the PowerCmp function

### PR DESCRIPTION
**Pull request by steven004**
This one play as the same function as previous one, but it might be more direct.

### CODE REVIEW

**Comment by steven004** [comment]
Please check #512 too. Just need one.

File: chain/types/blockheader.go [hunk 1]
```diff
@@ -166,24 +166,25 @@ func PowerCmp(eproof ElectionProof, mpow, totpow BigInt) bool {
...
+	// rhs = minerPower * 2^256
+	// rhs = minerPower << 256
 	rhs := new(big.Int).Lsh(mpow.Int, 256)
 	rhs = rhs.Mul(rhs, blocksPerEpoch.Int)
-	rhs = rhs.Sub(rhs, mpow.Int)
-
-	return lhs.Cmp(rhs) == -1
+	
+	// return true if lhs is less than or equal to rhs
+	return lhs.Cmp(rhs) < 1
```
Comment (magik6k) [inline chain/types/blockheader.go]:
```suggestion
	return lhs.Cmp(rhs) <= 0
```
(Basically the idea here is that `bigA.Cmp(bigB) [op] 0` == `bigA [op] bigB` (`[op]` being a compassion operation))

**Comment by whyrusleeping** [comment]
FYI @sternhenri @jbenet @nicola

**Comment by steven004** [comment]
> This is actually a different function, with this implementation the function in essence is:
> `h(vrfout) * totalPower < e * minerPower * 2^256`
> Which can be computed in easier way.

This one is quite good. It is equivalent to 
`(h(vrfout) + 1) * totalPower <= e * minerPower * 2^256`  while all of the elements are integers and in 2^256 space. let me make the change.

**Pull request merged**